staging/src/k8s.io/dynamic-resource-allocation/cel/compile.go

```diff
@@ -122,18 +130,35 @@ func (c compiler) CompileCELExpression(expression string, envType environment.Ty
 		return resultError("unexpected compilation error: "+err.Error(), apiservercel.ErrorTypeInternal)
 	}
 	prog, err := env.Program(ast,
+		// cel.CostLimit is also set to the VAP PerCallLimit as part of
+		// the base environment.
+		//
+		// This call here should override that. In practice it shouldn't
+		// matter because the limits are the same.
+		cel.CostLimit(resourceapi.CELSelectorExpressionMaxCost),
+		cel.CostTracking(estimator),
```

Comment:
It seems duplicated here since we have it set in https://github.com/kubernetes/kubernetes/blob/e673417529d9d2d2dd04f28f8437cbe1c403dfbe/staging/src/k8s.io/apiserver/pkg/cel/environment/base.go#L190 when true /* strictCost */) passed.

Reply:
That wasn't obvious. I agree, this can be removed.